api/v1alpha1/secretagentconfiguration_types.go

```diff
@@ -191,3 +254,40 @@ type Node struct {
 	AliasConfig  *AliasConfig  `yaml:"aliasConfig,omitempty"`
 	Value        []byte        `yaml:"value,omitempty"`
 }
+
+// Key Config Type Strings
+const (
+	TypeLiteral      KeyConfigType = "literal"
+	TypePassword     KeyConfigType = "password"
+	TypePrivateKey   KeyConfigType = "privateKey"
+	TypePublicKeySSH KeyConfigType = "publicKeySSH"
+	TypeCA           KeyConfigType = "ca"
+	TypeCAPrivateKey KeyConfigType = "caPrivateKey"
+	TypeCACopy       KeyConfigType = "caCopy"
+	TypePKCS12       KeyConfigType = "pkcs12"
+	TypeJCEKS        KeyConfigType = "jceks"
+	TypeJKS          KeyConfigType = "jks"
+	TypeTrustStore   KeyConfigType = "truststore"
+)
+
+// AliasConfigType Specifies which alias config type to use
+// +kubebuilder:validation:Enum=caCopy;keyPair;hmacKey;aesKey
+type AliasConfigType string
+
+// Alias Config Type Strings
+const (
+	TypeCACopyAlias AliasConfigType = "caCopy"
+	TypeKeyPair     AliasConfigType = "keyPair"
+	TypeHMACKey     AliasConfigType = "hmacKey"
+	TypeAESKey      AliasConfigType = "aesKey"
+)
+
+// Algorithm Specifies which keystore algorithm to use
+// +kubebuilder:validation:Enum=ECDSAWithSHA256;SHA256WithRSA
+type Algorithm string
+
+// Algorithm strings
+const (
+	ECDSAWithSHA256 Algorithm = "ECDSAWithSHA256"
```

Comment:
I guess this is OK - but I'd almost prefer the algo is a free format string that can be passed down to the command? 

i.e. how can this be extended with a new algo without having to recompile and update the CRD

Reply:
Wouldn't we require to update the operator code to support the new algorithm anyway? We can just update the `Enum` at that point. ;) Yes, this would change the CRD behind the scenes, but wouldn't hurt current users.

I personally like that validation of the field happens at the API level and we don't have to code that up in the webhook.